lib/dicthf.gi

```diff
@@ -104,6 +104,26 @@ local f,n,bytelen,data,qq,i;
              end;
 
     fi;
+  elif n > 100000 and Characteristic( f ) < n then
+    # large field, view it as an extension of its prime field
+    # in order to avoid writing down all of its elements
+    if Size( LeftActingDomain( f ) ) <> Characteristic( f ) then
+      f:= AsField( PrimeField( f ), f );
+    fi;
+    b:= Basis( f );
+    s:= LeftActingDomain( f );
+    sl:= AsSSortedList( s );
+    nn:= Size( s );
+    return function( v )
+      local sy, x, c;
+      sy:= 0;
+      for x in v do
+        for c in Coefficients( b, x ) do
+          sy:= nn*sy + ( Position( sl, c ) - 1 );
+        od;
```

Comment:
Isn't this what `NumberFFVector` is for? I.e. can't we do
```suggestion
        sy:= Size(f)*sy + NumberFFVector( c, nn );
```

Reply:
Yes, thanks for the hint.